### PR TITLE
Issue #1202: Improve setting IntegerObject values

### PR DESCRIPTION
**Pull request by elgorwi**
- Check if value for IntegerObject is in 32-bit range
- Raise ValueError for out of range values
- Add tests for the supported range and out of range

### CODE REVIEW

File: tests/test_cases/issue_134/test_integers.py [hunk 1]
```diff
@@ -16,12 +16,29 @@ def test_integer(dut):
     """
```
Comment (ktbarrett) [inline tests/test_cases/issue_134/test_integers.py:16]: Could you move both of the tests in this file into test_cocotb/test_handle, then delete the entire issue_134 directory?

File: cocotb/handle.py [hunk 2]
```diff
@@ -778,6 +783,11 @@ def _set_value(self, value, call_sim):
                 "Unsupported type for integer value assignment: {} ({!r})"
                 .format(type(value), value))
 
+        if value < INT_MIN or value > INT_MAX:
```
Comment (ktbarrett) [inline cocotb/handle.py:786]:
```suggestion
        if value < _INT_MIN or value > _INT_MAX:
```

File: tests/test_cases/issue_134/test_integers.py (hunk 1 above)
Comment (elgorwi) [reply tests/test_cases/issue_134/test_integers.py]: OK, also moved the "double" tests.

File: cocotb/handle.py [hunk 3]
```diff
@@ -47,6 +47,8 @@
 # Only issue a warning for each deprecated attribute access
 _deprecation_warned = set()
 
+INT_MAX = 2**31-1
+INT_MIN = -2**31
```
[the remark this answers is not in the document]
Comment (elgorwi) [reply cocotb/handle.py]: OK

File: cocotb/handle.py (hunk 2 above)
Comment (elgorwi) [reply cocotb/handle.py]: OK

**Comment by elgorwi** [comment]
Closing this in favour of #1742
